Ng20 upgrade with signal migrations

Langstra opened the pull request:
This PR does the following things:

- Upgrade to Angular 20
- Migrate to new control flow syntax
- Migrate to use input signals
- Migrate to use output functions instead of @Output decorator
- Migrate to use signal queries (viewChild, contentChildren, etc.)
- Upgrade to konva 9.2.22

chuma2150 commented:
Nice. Thanks. 
would expect peerDeps to be updated though

`  "peerDependencies": {
    "@angular/common": "^18.1.0",
    "@angular/core": "^18.1.0",
    "konva": "^9.3.1" `

Langstra commented:
> Nice. Thanks. would expect peerDeps to be updated though
> 
> ` "peerDependencies": { "@angular/common": "^18.1.0", "@angular/core": "^18.1.0", "konva": "^9.3.1"`

Thanks, good spot. I updated the peerDeps and also removed the tslib dependency. It is not used for what I can see. The demo still works after removing tslib.
